### PR TITLE
feat: Refactor/tracking

### PR DESCRIPTION
**Pull request by lmouhib**
Issue #, if available:

Description of changes: Added tracking capability for constructs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: core/src/common/context-options.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 export enum ContextOptions {
-  DISABLE_CONSTRUCTS_DEPLOYMENT_TRACKING = '@aws-analytics-reference-architecture/disableConstructsDeploymentTracking'
-}
\ No newline at end of file
+  DISABLE_CONSTRUCTS_DEPLOYMENT_TRACKING = '@aws-analytics-reference-architecture/disableConstructsDeploymentTracking',
+  EMR_EKS_TRACKING_ID = 'uksb-1sljh1bin',
+  DATA_ENG_PLATFORM_ID = 'uksb-1sljh1big',
```
Comment (vgkowski) [inline core/src/common/context-options.ts:4]: can we use the same code for EMR on EKS and Data_eng_platform? So we can keep one of them for future use

File: core/src/common/context-options.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+  DATA_LAKE_ID = 'uksb-1sljh1bja',
+  REPLAYER= 'uksb-1sljh1bgi',
```
Comment (vgkowski) [inline core/src/common/context-options.ts:6]: I would change to DATA_GENERATOR so it can be used to all the generator constructs

**Review by vgkowski — CHANGES_REQUESTED** [review]
Some constructs are missing codes like:

- AraBucket, DataLakeCatalog and DataLakeExporter should use the DATA_LAKE_ID
- DataGenerator should use the DATA_GENERATOR
- Can we have a code for all the helpers? (GlueDefaultRole, AthenaDemoSetup...)

File: core/src/common/context-options.ts (hunk 1 above)
Comment (lmouhib) [reply core/src/common/context-options.ts:4]: I would keep DATA_ENG_PLATFORM_ID and EMR_EKS_TRACKING_ID. The EMR_EKS_TRACKING_ID can be used without the DATA_ENG and we would want to keep distinct usage statistics for both.

Comment (lmouhib) [reply core/src/common/context-options.ts:4]: For AraBucket we will include it once cdk-cleaning is merged with main. 

For DataGenerator it is going to be deprecated soon in the next releases, and we will be only using the replayer, should we still track it?

Comment (vgkowski) [reply core/src/common/context-options.ts:4]: I wanted to be frugal on codes but no prob if we can have more. You are right about DataGenerator

**Review by vgkowski — DISMISSED** [review]

**Pull request merged**